### PR TITLE
Fix single program publish bug

### LINKED ISSUE #6194: Unable to Publish Programs With Certain Characters in Name
----
This only occurs when the `UNIVERSAL_QUESTIONS` flag is enabled.
----

Seattle's database has been around quite a while. As such some older data does not necessarily conform to newer validation requirements as the UI now requires.

I've found the older program "Gold Card (HSD)" which I put into draft mode on my local machine and the test server generate javascript errors and prevent `Publish all drafts` and `Publish` working for this program. Other programs that don't have this text can be published individually.

I tried swapping the modal id from using `program.adminName` to `program.slug` but that caused the publish button to disappear from the screen.

# Javascript Error

image

# Database Date

image

### PR DESCRIPTION
**Pull request by rockycodes**
### Description

When the Universal Questions feature flag is turned on, single program publish doesn't work with program admin names that include invalid characters. These characters are no longer allowed to be saved as `adminName` but this breaks with legacy data which impacts at least Seattle.

This fix changes the logic to reference `program.slug` instead of `program.adminName` when creating the reference name for the modal. `program.slug` is a [created from the admin name](https://github.com/civiform/civiform/blob/5e665380c10d7db32b1e8f1abdc264ada2455f68/server/app/services/program/ProgramDefinition.java#L495) with special characters stripped out.

#### General

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

### Instructions for manual testing

If instructions are needed for manual testing by reviewers, include them here.

1. Login as an admin and create a program
2. From the admin program index page, attempt to publish that one program
3. Program should publish successfully

Also try this with legacy data that includes invalid characters in `adminName` (Eg. "Gold Card (HSD)")

### Issue(s) this completes

Fixes #6194 ;

### CODE REVIEW

File: browser-test/src/admin_program_list.test.ts [hunk 1]
```diff
@@ -173,12 +173,12 @@ describe('Program list page.', () => {
     await adminPrograms.addQuestionFromQuestionBank(nameQuestion)
 
     await adminPrograms.gotoAdminProgramsPage()
-    await page.click('#program-one-publish-modal-button')
-    expect(await page.innerText('#program-one-publish-modal')).toContain(
+    await page.click('#publish-modal-program-one-button')
```
Comment (gwendolyngoetz) [inline browser-test/src/admin_program_list.test.ts:176]: Non-blocking: Consider putting all the various `#publish-modal-xxx` and `#publish-modal-xxx-button` strings into a variable.

File: server/app/views/admin/programs/ProgramIndexView.java [hunk 2]
```diff
@@ -263,7 +263,7 @@ private ImmutableList<Modal> buildPublishSingleProgramModals(
                           "flex", "flex-col", StyleUtils.responsiveMedium("flex-row"), "py-4");
 
               return Modal.builder()
-                  .setModalId(program.adminName() + "-publish-modal")
+                  .setModalId("publish-modal-" + program.slug())
```
Comment (gwendolyngoetz) [inline server/app/views/admin/programs/ProgramIndexView.java:266]: Blocking: Recommend creating a method you can pass a program to that will build the modal ID so it doesn't get out of sync between the spots you need it.

**Review by gwendolyngoetz — CHANGES_REQUESTED** [review]

File: server/app/views/admin/programs/ProgramIndexView.java (hunk 2 above)
Comment (rockycodes) [reply server/app/views/admin/programs/ProgramIndexView.java]: good idea. done.

File: browser-test/src/admin_program_list.test.ts (hunk 1 above)
Comment (rockycodes) [reply browser-test/src/admin_program_list.test.ts]: done

**Review by gwendolyngoetz — APPROVED** [review]

**Pull request merged**